insane.go

```diff
@@ -144,11 +143,28 @@ type StrictNode struct {
 }
 
 type decoder struct {
-	id        int
-	buf       []byte
-	root      Root
-	nodePool  []*Node
-	nodeCount int
+	id       int
+	buf      []byte
+	root     Root
+	nodePool []Node
+}
+
+func newDecoder() *decoder {
+	return &decoder{nodePool: make([]Node, 0, 16)}
```

Comment:
Initializing capacity using the StartNodePoolSize variable did not decrease/increase performance, but it does affect the memory consumed, so it was decided to remove this variable. 

I understand that this will break backward compatibility for some applications (applications will have to remove use of the StartNodePoolSize variable).